doc/tutorials/nvmx_virtualization.rst

```diff
@@ -0,0 +1,289 @@
+
+Secondly, make sure the ``bootargs`` setting doesn't include any ``console=xxx``
+test, for example:
+
+   .. code-block:: xml
+      :emphasize-lines: 4,5
+
+      <board_private>
+        <rootfs>/dev/sda3</rootfs>
+        <bootargs>
+        rw rootwait consoleblank=0 no_timer_check quiet loglevel=3
+        i915.nuclear_pageflip=1 hvlog=2M@0xe00000 memmap=0x200000$0xe00000
+        </bootargs>
+      </board_private>
+
+After ACRN is build, validate this configuration from the generated code.
+For example, check this file if you build a SDC scenario:
+``acrn-hypervisor/build/hypervisor/configs/scenarios/sdc/misc_cfg.h``,
+the ``SOS_CONSOLE`` should be empty and ``SOS_BOOTARGS_DIFF`` should not
+include any ``console=xxx`` text.
+
+   .. code-block:: xml
+      :emphasize-lines: 2,6
+
+      #define SOS_ROOTFS                    "root=/dev/sda3 "
+      #define SOS_CONSOLE                   " "
+      #define SOS_COM1_BASE                 0x2F8U
+      #define SOS_COM1_IRQ                  3U
+
+      #define SOS_BOOTARGS_DIFF             "rw rootwait consoleblank=0
+            no_timer_check quiet loglevel=3 i915.nuclear_pageflip=1
+            hvlog=2M@0xe00000 memmap=0x200000$0xe00000 maxcpus=4 "
+
+- Configure system level features:
+
+  - Edit ``cpu_affinity`` to assign pCPU IDs to run the L1 guest. If you are
+    using debug build and need the hypervisor console, please don't assign
+    pCPU0 to the VM that enables nVMX
+  - Edit ``guest_flag`` to enable ``GUEST_FLAG_NVMX_ENABLED`` on the VM that
+    you want to run nested guests on
+  - Edit ``guest_flag`` to enable ``GUEST_FLAG_LAPIC_PASSTHROUGH`` on this
+    nVMX enabled VM
+
+   .. code-block:: xml
+
+      <vm id="0">
+        <vm_type>SOS_VM</vm_type>
+        <name>ACRN SOS VM</name>
+        <cpu_affinity>
+          <pcpu_id>1</pcpu_id>
+          <pcpu_id>2</pcpu_id>
+          <pcpu_id>3</pcpu_id>
+        </cpu_affinity>
+        <guest_flags>
+          <guest_flag>GUEST_FLAG_NVMX_ENABLED</guest_flag>
+          <guest_flag>GUEST_FLAG_LAPIC_PASSTHROUGH</guest_flag>
+        </guest_flags>
+
+- Build with the XML configuration, refer to :ref:`getting-started-building`.
+
+
+Launch nested guests
+********************
+
+Since we disable serial console on Service OS if nVMX is enabled, you can log
+in the Service OS with SSH.
+
+Run QEMU on L1 guest to launch nested guest. Please note that ``-cpu host``
+is needed to launch nested guests, and currently ``-nographics`` is required
+to be able to run nested guests reliably.
+
+   .. code-block:: bash
+      :emphasize-lines: 2,3,4
+
+      qemu-system-x86_64 \
+        -enable-kvm \
+        -cpu host \
+        -nographic \
+        -m 2G \
+        -smp 2 \
+        -hda /root/ubuntu-20.04.qcow2 \
+        -net nic,macaddr=00:16:3d:60:0a:80 -net tap,script=/etc/qemu-ifup
+
+If the nested VM is launched successfully, you can have a working console:
+
+   .. code-block:: bash
+
+      Ubuntu 18.04.2 LTS ubuntu_vm ttyS0
```

Comment:
Minor thing but this indicates Ubuntu 18.04 while the name of the `qcow2` file indicates a Ubuntu 20.04 OS. Can we make these consistent?